proto/cosmos/tx/v1beta1/tx.proto

```diff
@@ -63,6 +63,33 @@ message SignDoc {
   uint64 account_number = 4;
 }
 
+// SignDocTextual is the type used for generating sign bytes for SIGN_MODE_TEXTUAL. It is designed to be serialized
+// as proto3 JSON following the rules defined Here: https://github.com/regen-network/canonical-proto3/blob/master/README.md#json.
+message SignDocTextual {
```

Comment:
Since it's not as textual as initially planned anymore, maybe we could call is `SIGN_MODE_JSON`?

Reply:
nit: Can we wrap this at 80 columns?

Reply:
up, the 2 comments are still relevant

Reply:
Unfortunately, we already added `SIGN_MODE_TEXTUAL` into our `SignMode` enum so that would be proto breaking. We can add a enum case for `SIGN_MODE_DIRECT_JSON` and not use `SIGN_MODE_TEXTUAL` or bump the proto version and change it. What do you think @amaurym ?

Reply:
I think it seems reasonable to add an additional `SIGN_MODE_DIRECT_JSON` that makes it clear we're using a canonical JSON representation of the protobuf bytes. We can then keep `SIGN_MODE_TEXTUAL` for that ICU-inspired human-readable message signing you mentioned in https://github.com/cosmos/cosmos-sdk/issues/6513#issuecomment-743344943.

Reply:
Agreed. I would also add some proto comments in the enum saying that `SIGN_MODE_TEXTUAL` is currently not supported